### ISSUE TITLE
Add ruby24 and trunk to appveyor?

### ISSUE THREAD

**Issue by MSP-Greg** [ruby/spec#465]
I've added 2.4 & trunk to appveyor.  tests are [here](https://ci.appveyor.com/project/MSP-Greg/spec), updated file is [here](https://github.com/MSP-Greg/spec/blob/f6642c4a3a21b106f3d2cb9c9f2791255687f254/appveyor.yml).

1. Would you like a PR?
2. 2.4 only, or 2.4 & trunk (can also add trunk as an allowable failure)?

FYI, appveyor has been asked to update ruby24 to 2.4.2.

**Comment by eregon** [comment]
Please create a PR, this sounds great!

For trunk, I see you use your own build. Will it be often updated?
The problem with ruby trunk on Travis CI is it often gets outdated and therefore starts failing, making it mostly useless.

My only concern is it takes longer for CI but macOS on Travis is so slow nowadays that it probably does not matter much.

**Comment by MSP-Greg** [comment]
@eregon 

> For trunk, I see you use your own build. Will it be often updated?

I call it my 'breakfast build'.  Short of a catastrophic failure/error, I update it daily.  Its come far enough along that I'm thinking of moving the build to appveyor, but I'm trying to see if I can set it up so one can use it for both local builds (using a git repo) and appveyor builds.

FYI, when I build, I run both `make test-spec` and `..\mspec\bin\mspec -rdevkit`.

This morning, `make test-spec` had two fails in WIN32OLE that I've never seen before, but `mspec` did not.  I'll check further, and based on that, push the appveyor.yml update later.

Thanks again.

**Comment by MSP-Greg** [comment]
@eregon 

I haven't had time to look at it, but I just squashed commits and re-ran tests in my fork.  The build this morning passed mspec
```
3551 files, 26041 examples, 202825 expectations, 0 failures, 0 errors, 0 tagged
```
But the recent [fork build results](https://ci.appveyor.com/project/MSP-Greg/spec/build/job/wtjbaeda80nbjv0i) showed
```
3548 files, 25984 examples, 202966 expectations, 6 failures, 0 errors, 0 tagged
```

Again, I haven't looked further...

**Comment by eregon** [comment]
This repo and MRI's copy in spec/ruby are only sync'd about once per month, so it's expected there are a few differences.

I think this can be closed since #468 is merged.

**Comment by MSP-Greg** [comment]
Benoit, thanks for all your work.

> This repo and MRI's copy in spec/ruby are only sync'd about once per month

Am I correct that, at present, changes are generally pushed from trunk to here?  Exceptions for PR's here, etc.

If you know (without taking any time to check), any idea what causes the difference in 2.4 and trunk numbers? 
```
2.4   - 3554 files, 26019 examples, 203409 expectations, 0 failures, 0 errors, 0 tagged
trunk - 3554 files, 25996 examples, 202980 expectations, 6 failures, 0 errors, 0 tagged
```

Call me AR, but I've always been bothered by test stats like this, since they don't give any info about 
why there are differences in examples and expectations.  Same issue with the ruby test-all suite...

Greg

**Comment by eregon** [comment]
> Am I correct that, at present, changes are generally pushed from trunk to here? Exceptions for PR's here, etc.

It's really mixed, there are PRs here but also changes from MRI, TruffleRuby and JRuby which each have a copy of this repo and synchronized about once a month here.

> Call me AR

What does "AR" mean? I don't know that abbreviation.

Some examples (`it` blocks) are only valid on executed versions due to the `ruby_version_is` guard, so it's expected to have different numbers for 2.4 and trunk.
Just after a synchronization, numbers with `make test-spec` and run directly against trunk should be identical, I'll check next time. But in between changes in MRI or here can change the numbers of course.

The last 3 numbers are the more useful ones, and `examples` to get a decent approximation of how much of ruby/spec is run and pass on other implementations.

**Comment by MSP-Greg** [comment]
> What does "AR" mean? I don't know that abbreviation.

Sorry, wasn't sure how common that was.  I previously was involved with 'high-reliability' system design.

—often used in nontechnical contexts to describe someone as extremely or excessively neat, careful, or precise

a*al-retentive

Thanks for the explanation re stats.  Re MRI, I've got results of trunk spec tests [here](https://msp-greg.github.io/file.mingw_test-all.html#test-spec-Failure-Error-Summary).  GitHub.io / GitHub Pages is a little slow today, I've got 22-Sept updates that aren't appearing yet...

**Comment by MSP-Greg** [comment]
@eregon

> For trunk, I see you use your own build. Will it be often updated?

The build is now a rolling build (every 12 hours) on Appveyor.

I looked over the CAPI files, and (with my limited c knowledge) it seemed that it was all core.  The build uses an updated OpenSSL library (1.1.0f) and a ruby specific GDBM package.  Since the CAPI files don't seem to link to any of that, the packages probably aren't required?

Also, current Appveyor environment uses GCC 6.30, but 7.2.0 is available.  At present, anyone installing MSYS2 will have gcc 7.2.0 without additional work.  FYI, MSYS2 is required for building extension gems when using ruby 2.4 + on windows.   It is included in the standard installation app.

 I've got a script included with the build that updates to 7.2.0, yesterday it took 1:10 to run.

Questions:

1. Should I update appveyor.yml to use the rolling build?
2. Run the 7.2.0 script, or stay with standard Appveyor environment?

I juts ran the update (with gcc 7.2.0) on my fork, [trunk results](https://ci.appveyor.com/project/MSP-Greg/spec/build/5/job/j2chxe38vbgq96ps) are the same as the [last commit](https://ci.appveyor.com/project/eregon/spec-x948i/build/728/job/ef1y37ism77c2fsc).  2.3 and 2.4 pass as before.

Thanks,

Greg
